include/measurement_kit/http/http.hpp

```diff
@@ -328,7 +327,7 @@ class Stream {
      */
     void on_connect(std::function<void(void)>&& fn) {
         connect_handler = fn;
-        connection->on_connect([this]() {
+        connection.on_connect([this]() {
```

Comment:
Reading more carefully the header file, since `connection` is a transport, nothing really changed in this case since before. That is, `connection` used to be `SharedPointer<Transport>`. Now `Transport` (the abstract thing) is `TransportInterface` and `Transport` contains a `SharedPointer<TransportInterface>`. So, at the end of the day, here what really changed is that using transport is easier since the template is hidden.